### PR TITLE
Added support for Patchouli Books

### PR DESCRIPTION
**Pull request by SimonFlapse**
* This allows the creating of in-game books for documentations
* Provided example with Block Breaker and Block Placer
* Replaces old Tech Reborn Manual

Moved PR from:
https://github.com/TechReborn/TechReborn/pull/2895

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -318,6 +328,7 @@ task renameCrowdin(type: Copy, dependsOn: ['crowdin', 'cleanCrowdin']){
 import groovy.json.JsonOutput
 import groovy.json.JsonSlurper
 import groovy.util.XmlSlurper
+import org.kohsuke.github.GHReleaseBuilder
```
Comment (Ayutac) [inline build.gradle:331]: why is the import order changed?

File: src/main/resources/data/techreborn/patchouli_books/techreborn/book.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+	"name": "patchouli.techreborn.guide.name",
+	"landing_text": "patchouli.techreborn.landing_text",
+	"model": "techreborn:manual",
+	"version": 1,
+	"show_progress": false,
+	"pause_game": false,
+	"dont_generate_book": true,
+	"custom_book_item": "techreborn:manual",
+	"i18n": true
```
Comment (Ayutac) [inline src/main/resources/data/techreborn/patchouli_books/techreborn/book.json:10]: No blocky font?

Comment (SimonFlapse) [reply src/main/resources/data/techreborn/patchouli_books/techreborn/book.json:10]: I'll probably add it, the font clips at certain UI scales right now.

File: build.gradle (hunk 1 above)
Comment (SimonFlapse) [reply build.gradle:331]: I'll revert that according to modmuss's review on the first PR

File: src/main/java/techreborn/items/ManualItem.java [hunk 3]
```diff
@@ -47,7 +47,15 @@ public ManualItem() {
 	@Override
 	public TypedActionResult<ItemStack> use(final World world, final PlayerEntity player, final Hand hand) {
 		if (player instanceof ServerPlayerEntity serverPlayerEntity) {
-			PatchouliAPI.get().openBookGUI(serverPlayerEntity, new Identifier("techreborn:techreborn"));
+            if (FabricLoader.getInstance().isModLoaded("patchouli")) { // TODO replace string by constant TechReborn.PATCHOULI_ID
+				PatchouliAPI.get().openBookGUI(serverPlayerEntity, new Identifier("techreborn:techreborn"));
+				// does 
+                // PatchouliAPI.get().openBookGUI(serverPlayerEntity, Registry.ITEM.getId(this));
+				// not work? Identifier techreborn:techreborn is a bad one btw
```
Comment (SimonFlapse) [inline src/main/java/techreborn/items/ManualItem.java:54]: It expects <[modid]>:<[bookname]> based on the patchouli reference guide.

I called the book techreborn (The name of the folder within the patchouli_books folder)

**Comment by Technici4n** [comment]
If you want to help with the documentation effort, please contribute to the wiki. A guidebook would lead to information duplication, and is therefore not going to be accepted.

**Pull request closed without merging**